compiler/rustc_const_eval/src/transform/validate.rs

```diff
@@ -677,7 +677,14 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
 
                 let kind = match parent_ty.ty.kind() {
                     &ty::Alias(ty::Opaque, ty::AliasTy { def_id, args, .. }) => {
-                        self.tcx.type_of(def_id).instantiate(self.tcx, args).kind()
+                        let ty = self.tcx.type_of(def_id).instantiate(self.tcx, args);
+                        // If the type we get is opaque, we want to normalize it.
+                        if ty.is_impl_trait() {
+                            let inner_ty = self.tcx.expand_opaque_types(ty).kind();
+                            inner_ty
+                        } else {
+                            ty.kind()
+                        }
```

Comment:
This does not seem right. Why aren't we just using the regular `normalize_erasing_regions` that other places use in this validator?